### PR TITLE
Code fixes

### PR DESCRIPTION
**Pull request by AntonStoeckl**
Hi Pablo,

here we go, my pull request with the code fixes.
I have added unit tests for the idToMongo() stuff and for sure ran all unit test to see if nothing else breaks.

Cheers, Anton

### CODE REVIEW

File: src/Mandango/Extension/Core.php [hunk 1]
```diff
@@ -658,6 +658,9 @@ private function initDefinitionsProcess()
      */
     private function globalInheritableAndInheritanceProcess()
     {
+        $inheritable = null;
+        $inheritableClass = null;
```
Comment (pablodip) [inline src/Mandango/Extension/Core.php:662]: What is failing without these initializations?

File: src/Mandango/Logger/LoggableMongoCollection.php [hunk 2]
```diff
@@ -18,7 +18,7 @@
  */
 class LoggableMongoCollection extends \MongoCollection
 {
-    private $db;
+    public $db;
```
Comment (pablodip) [inline src/Mandango/Logger/LoggableMongoCollection.php:21]: Why public?

File: src/Mandango/Logger/LoggableMongoCursor.php [hunk 3]
```diff
@@ -87,7 +87,7 @@ public function rewind()
     {
         $this->logQuery();
 
-        return parent::rewind();
+        parent::rewind();
```
Comment (pablodip) [inline src/Mandango/Logger/LoggableMongoCursor.php:90]: ?

**Comment by pablodip** [comment]
Again, this PR includes different fixes. There should be a PR for each different fix :)

File: src/Mandango/Extension/Core.php (hunk 1 above)
Comment (AntonStoeckl) [inline src/Mandango/Extension/Core.php:662]: nothing so far, but both variables might be undefined when they are used, if this if branch is not hit:

if ($this->configClasses[$inheritanceClass]['inheritable']) {

File: src/Mandango/Logger/LoggableMongoCollection.php (hunk 2 above)
Comment (AntonStoeckl) [inline src/Mandango/Logger/LoggableMongoCollection.php:21]: It's derived from \MongoCollection and there:

```
/**
 * @var MongoDB
 */
public $db = NULL ;
```

So the access must be public or weaker in derived classes. Again, this is not breaking anything.

File: src/Mandango/Logger/LoggableMongoCursor.php (hunk 3 above)
Comment (AntonStoeckl) [inline src/Mandango/Logger/LoggableMongoCursor.php:90]: \MongoCursor::rewind returns noting (void), so the child class should not return anything.

**Comment by AntonStoeckl** [comment]
I'll close this and email you about the MongoID thing as I have run into some questions about how the _id id handled in Mandango generally.

Regards, Anton

**Pull request closed without merging**